azure-iot-device/azure/iot/device/iothub/aio/async_clients.py

```diff
@@ -350,6 +384,26 @@ async def notify_blob_upload_status(
         await handle_result(callback)
         logger.info("Successfully notified blob upload status")
 
+    @property
+    def on_method_request_received(self):
+        return self._on_method_request_received
+
+    @on_method_request_received.setter
+    def on_method_request_received(self, value):
+        self._validate_receive_handler_setter()
+        # Enable the feature if necessary
+        if value is not None and not self._mqtt_pipeline.feature_enabled[constant.METHODS]:
+            loop = asyncio.get_event_loop()
```

Comment:
>loop = asyncio.get_event_loop() [](start = 12, length = 31)

huh, so I guess we can't have async setters.  Random comment: Longer term, we'll probably need to scrub the code to remove all `import asyncio` in favor of asyncio_compat.

Reply:
Async setters aren't supported by the @property decorator, no. There's actually a bit of an issue that this creates that I want to address before we start exposing these to users.

---
In reply to: [454511914](https://github.com/Azure/azure-iot-sdk-python/pull/620#discussion_r454511914) [](ancestors = 454511914)